spec/nativeTemplateEngineBehaviors.js

```diff
@@ -1,100 +1,117 @@
+
         it('can display static content from regular DOM element', function () {
+            var testDivTemplate = ensureNodeExistsAndIsEmpty("testDivTemplate");
             window.testDivTemplate.innerHTML = "this is some static content";
-            ko.renderTemplate("testDivTemplate", null, null, window.templateOutput);
-            expect(window.templateOutput).toContainHtml("this is some static content");
+            ko.renderTemplate("testDivTemplate", null, null, window.testNode);
+            expect(window.testNode).toContainHtml("this is some static content");
         });
 
         it('can fetch template from regular DOM element and data-bind on results', function () {
-            window.testDivTemplate.innerHTML = "name: <div data-bind='text: name'></div>";
-            ko.renderTemplate("testDivTemplate", { name: 'bert' }, null, window.templateOutput);
-            expect(window.templateOutput).toContainHtml("name: <div data-bind=\"text: name\">bert</div>");
+            var testDivTemplate = ensureNodeExistsAndIsEmpty("testDivTemplate");
+            testRenderTemplate(testDivTemplate, "testDivTemplate");
         });
 
         it('can fetch template from <script> elements and data-bind on results', function () {
-            window.testScriptTemplate.text = "name: <div data-bind='text: name'></div>";
-            ko.renderTemplate("testScriptTemplate", { name: 'bert' }, null, window.templateOutput);
-            expect(window.templateOutput).toContainHtml("name: <div data-bind=\"text: name\">bert</div>");
+            var testScriptTemplate = ensureNodeExistsAndIsEmpty("testScriptTemplate", "script", "text/html");
+            testRenderTemplate(testScriptTemplate, "testScriptTemplate", "text");
         });
 
         it('can fetch template from <textarea> elements and data-bind on results', function () {
-            var prop = (typeof window.testTextAreaTemplate.innerText !== "undefined") ? "innerText" : "textContent";
-            window.testTextAreaTemplate[prop] = "name: <div data-bind='text: name'></div>";
-            ko.renderTemplate("testTextAreaTemplate", { name: 'bert' }, null, window.templateOutput);
-            expect(window.templateOutput).toContainHtml("name: <div data-bind=\"text: name\">bert</div>");
+            var testTextAreaTemplate = ensureNodeExistsAndIsEmpty("testTextAreaTemplate", "textarea"),
+                prop = (typeof testTextAreaTemplate.innerText !== "undefined") ? "innerText" : "textContent";
+            testRenderTemplate(testTextAreaTemplate, "testTextAreaTemplate", prop);
+        });
+
+        it('can fetch template from <template> elements and data-bind on results', function () {
+            var testTemplateTemplate = ensureNodeExistsAndIsEmpty("testTemplateTemplate", "template");
+            testRenderTemplate(testTemplateTemplate, "testTemplateTemplate");
         });
     });
 
     describe('Anonymous templates', function () {
+        beforeEach(jasmine.prepareTestNode);
```

Comment:
Doesn't make much real difference, but `prepareTestNode` is getting called already for each test in the outer `beforeEach`.